pkg/acsengine/engine.go

```diff
@@ -984,6 +987,27 @@ func getDataDisks(a *api.AgentPoolProfile) string {
 	return buf.String()
 }
 
+func getEtcdDisk(orchestratorType api.OrchestratorType, m *api.MasterProfile) string {
+	if orchestratorType != api.Kubernetes {
+		return ""
+	}
+	var buf bytes.Buffer
+	buf.WriteString("\"dataDisks\": [\n")
+	etcdDisks := `            {
+              "createOption": "Empty",
+              "diskSizeGB": "%d",
+              "lun": %d,
+              "name": "[concat(variables('masterVMNamePrefix'), copyIndex(),'-etcddisk')]",
+              "vhd": {
+                "uri": "[concat(reference(concat('Microsoft.Storage/storageAccounts/',variables('masterStorageAccountName')),variables('apiVersionStorage')).primaryEndpoints.blob,'vhds/',    variables('masterVMNamePrefix'),copyIndex(),'-etcddisk.vhd')]"
+               }
+            }`
+
+	buf.WriteString(fmt.Sprintf(etcdDisks, 1, 0))
```

Comment:
Updated. 

I don't know the size of etcd in a 5000 cluster. But for a 3 master, 2 agent cluster its 3MB. I anticipate this will grow to some degree over time as well. Even if the etcd size grows linearly with each new node added we should be within the 128 GB limit. Though it would be interesting to find out what 1200 node cluster shows about etcd size change..